src/nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -1078,10 +1078,7 @@ def add_units_table(
     else:
         checked_sorting = sorting
 
-    sampling_frequency = checked_sorting.get_sampling_frequency()
-    if sampling_frequency is None:
-        raise ValueError("Writing a SortingExtractor to an NWBFile requires a known sampling frequency!")
-
+    write_table_first_time = False
```

Comment:
What about

```
write_table_first_time = units_table_name not in ecephys_mod.data_interfaces if write_in_processing_module else nwbfile.units is None
```

Saves you two lines below that are in the double-nested `if` conditions

Reply:
It annoys me that it is a little bit too long and black decomposes it. On the other hand, besides the line saving, we get to handle the whole logic of the assignation in one line instead of having it distributed over the control structures below (hence better for short-memory). I will change it.

I wish we could have the definition closer to where it is in fact used but this version needs to be almost at the beginning.

Reply:
> Oh, the problem is that this requires you to have previously defined ecephys_mod

Ah, right, dang.

But yeah I generally prefer to try to do these kinds of things in a single clear definition rather than nested logic which requires both mental tracking of nested logic and eye-tracking to navigate code.